dali/python/nvidia/dali/experimental/dynamic/_op_builder.py

```diff
@@ -496,13 +607,15 @@ def _infer_device():
                     device.device_type = "mixed"
 
         # Get or create the operator instance that matches the arguments
+        # Exclude _random_state from caching key since it changes on every call
+        caching_call_arg_names = tuple(k for k in call_args.keys() if k != "_random_state")
         with nvtx.annotate(f"get instance {op.op_name}", domain="op_builder"):
             op_inst = op.get(
                 max_batch_size=max_batch_size,
                 name=None,
                 device=device,
                 num_inputs=len(inputs),
-                call_arg_names=tuple(call_args.keys()),
+                call_arg_names=caching_call_arg_names,
```

Comment:
Why not just pass `rng` unchanged here and handle it in `__call__`, which we already do?